packages/dds/tree/src/test/feature-libraries/modular-schema/modularChangeFamily.spec.ts

```diff
@@ -1048,6 +1064,195 @@ describe("ModularChangeFamily", () => {
 		});
 	});
 
+	describe("adds missing builds", () => {
+		const aMajor = mintRevisionTag();
+		const a1 = { major: aMajor, minor: 1 };
+		const a2 = { major: aMajor, minor: 2 };
+		const bMajor = mintRevisionTag();
+		const b1 = { major: bMajor, minor: 1 };
+		const cMajor = mintRevisionTag();
+		const c1 = { major: cMajor, minor: 1 };
+		const u1 = { minor: 1 };
+
+		const node2 = singleJsonCursor(2);
+		const node2Chunk = treeChunkFromCursor(node2);
+		const node3 = singleJsonCursor(3);
+		const node3Chunk = treeChunkFromCursor(node3);
+
+		const nodesArray: [DeltaDetachedNodeId, TreeChunk][] = [
+			[a1, node1Chunk],
+			[a2, node2Chunk],
+			[b1, node3Chunk],
+		];
+		const nodeMap = nestedMapFromFlatList(
+			nodesArray.map(([{ major, minor }, chunk]) => [major, minor, chunk]),
+		);
+
+		const getDetachedNode = ({ major, minor }: DeltaDetachedNodeId) => {
+			return tryGetFromNestedMap(nodeMap, major, minor);
+		};
+
+		it("preserves relevant builds that are present in the input", () => {
+			const input: ModularChangeset = {
+				fieldChanges: new Map([]),
+				builds: new Map([
+					[undefined, new Map([[brand(1), node2Chunk]])],
+					[aMajor, new Map([[brand(2), node2Chunk]])],
+					[cMajor, new Map([[brand(1), node3Chunk]])],
+				]),
+			};
+
+			const expected: ModularChangeset = {
+				fieldChanges: new Map([]),
+				builds: new Map([
+					[undefined, new Map([[brand(1), node2Chunk]])],
+					[
+						aMajor,
+						new Map([
+							[brand(1), node1Chunk],
+							[brand(2), node2Chunk],
+						]),
+					],
+					[cMajor, new Map([[brand(1), node3Chunk]])],
+				]),
+			};
+
+			const withBuilds = addMissingBuilds(makeAnonChange(input), getDetachedNode, [
+				u1,
+				a1,
+				a2,
+				c1,
+			]);
+			assert.deepEqual(withBuilds, expected);
+		});
+
+		it("preserves irrelevant builds that are present in the input", () => {
+			const input: ModularChangeset = {
+				fieldChanges: new Map([]),
+				builds: new Map([
+					[undefined, new Map([[brand(1), node2Chunk]])],
+					[aMajor, new Map([[brand(2), node2Chunk]])],
+					[cMajor, new Map([[brand(1), node3Chunk]])],
+				]),
+			};
+
+			const expected: ModularChangeset = {
+				fieldChanges: new Map([]),
+				builds: new Map([
+					[undefined, new Map([[brand(1), node2Chunk]])],
+					[
+						aMajor,
+						new Map([
+							[brand(1), node1Chunk],
+							[brand(2), node2Chunk],
+						]),
+					],
+					[cMajor, new Map([[brand(1), node3Chunk]])],
+				]),
+			};
+
+			const withBuilds = addMissingBuilds(makeAnonChange(input), getDetachedNode, [a1]);
+			assert.deepEqual(withBuilds, expected);
+		});
+
+		describe("attemps to add relevant builds that are missing from the input", () => {
```

Comment:
That was probably my bad
```suggestion
		describe("attempts to add relevant builds that are missing from the input", () => {
```